samples/benchmarks/coremark/README.rst

```diff
@@ -0,0 +1,146 @@
+
+You can select the following memory allocation methods:
+
+* The :kconfig:option:`CONFIG_COREMARK_MEMORY_METHOD_STACK` Kconfig option uses the main thread stack for data allocation and is enabled by default.
+  You can change the main stack size by setting the :kconfig:option:`CONFIG_MAIN_STACK_SIZE` Kconfig option.
+* The :kconfig:option:`CONFIG_COREMARK_MEMORY_METHOD_MALLOC` Kconfig option uses Zephyr :c:func:`k_malloc\k_free` functions to allocate the memory in the heap.
+  You can adjust the heap size with the :kconfig:option:`CONFIG_HEAP_MEM_POOL_SIZE` Kconfig option.
+* The :kconfig:option:`CONFIG_COREMARK_MEMORY_METHOD_STATIC` option allows data allocation in the RAM as a regular static variable.
+  In this case you do not need to be aware of the main thread stack size or heap memory pool size.
+
+CoreMark can also be executed in the multithread mode.
+To specify a number of threads, use the :kconfig:option:`CONFIG_COREMARK_THREADS_NUMBER` Kconfig option.
+You can configure up to four threads.
+In the multithread mode, CoreMark is executed in all threads simultaneously.
+By default, the :kconfig:option:`CONFIG_TIMESLICE_SIZE` Kconfig option is set to 10 ms which imitates RTOS multithread usage.
+However, in the final report the thread execution is combined.
+In the multithread mode, the :kconfig:option:`CONFIG_COREMARK_DATA_SIZE` Kconfig option is allocated for each thread separately.
+Only the :kconfig:option:`CONFIG_COREMARK_MEMORY_METHOD_STACK` and :kconfig:option:`CONFIG_COREMARK_MEMORY_METHOD_MALLOC` memory methods can be used with multiple threads.
+In case of the :kconfig:option:`CONFIG_COREMARK_MEMORY_METHOD_STACK` Kconfig option, the data for all threads are allocated to the main thread stack.
+
+Building and running
+********************
+
+When running the benchmark, an extra build flag (:kconfig:option:`CONFIG_COMPILER_OPT`) is set to ``-O3`` to achieve the best CoreMark results.
+
+.. code-block:: sh
+
+   west build -b nrf5340dk_nrf5340_cpuapp samples/benchmark/coremark
+   west flash
+
+After flashing, messages describing the benchmark state will appear in the console.
+
+Testing
+=======
+
+After programming the sample to your development kit, complete the following steps to test it:
+
+1. Connect to the development kit with a terminal emulator (for example, PuTTY).
+#. Reset your development kit.
+#. To start the test, press the button assigned to the respective core.
+   To see the assignment, refer to the :ref:`coremark_user_interface` section.
+   All target cores work independently from each other.
+   It is possible to run the benchmark on several cores simultaneously.
+   Measurements running on a core will be indicated by the corresponding LED.
+   If the :kconfig:option:`COREMARK_MODE_FLASH_AND_RUN` Kconfig option is enabled, the measurement will launch on the application's start automatically on all available cores.
```

Comment:
Shouldn't this be broken into several steps and/or scenarios? @peknis ?

Reply:
Maybe some bullet points could make this a bit clearer.